### DISCUSSION TITLE
Since AdGuard moved to Mv3, is it an idea to use AdGuard's open source tsURLfilter, scriptlets and extendedCSS to create an uBO Mv3 version?

### DISCUSSION THREAD

**Discussion by Kees1958** [uBlockOrigin/uBlock-issues#3388] [Ideas] [answered]
As the title menstions, uBOL already provides great functionality using DNR, to provide full uBO functionality uBO could use the opensource functionality of AdGuard (which is now fully Mv3 compatible).

Google's performance performance argument seems to be proven by speedometer 3 runs (AG Mv3 versus uBO Mv2 in Firefox).

**Comment by garry-ut99** [comment]
It was decided long time ago already, that there will not be a full uBO Mv3 version.
There exists uBOL instead, which never was supposed to be a full replacement of uBO:
- https://github.com/uBlockOrigin/uBOL-home/discussions/153#discussioncomment-10105188

<hr>

> Kees1958 :  AdGuard's open source URLfilter

It seems you mean this: https://github.com/AdguardTeam/urlfilter

<hr>

> Kees1958 : Google's performance performance argument seems to be proven by speedometer 3 runs (AG Mv3 versus uBO Mv2 in Firefox).

However, in another benchmark: [JetStream2](https://browserbench.org/JetStream2.0/), uBO despite still losing in Firefox, it suprisingly wins in Chromium, see: https://github.com/uBlockOrigin/uBlock-issues/discussions/3383#discussioncomment-10726971

**Reply by gorhill** [reply]
> It seems you mean this: <https://github.com/AdguardTeam/urlfilter>

It's a golang project which purpose appears to be the creation of a local proxy to filter network request, and with very little repo activity. This has nothing to do with browser extensions, so if this is what @Kees1958 really referenced, I don't know why this would be brought into a topic about browser extensions -- explanations as to why are welcome.

**Reply by Yuki2718** [reply, marked as answer]
I think he actually meant https://github.com/AdguardTeam/tsurlfilter

**Reply by Kees1958** [reply]
@Yuki2718 you are correct, it was a typo, thanks for correcting

**Comment by gorhill** [comment]
If you prefer AdGuard, why not just use AdGuard?

**Reply by Kees1958** [reply]
Due to geo political issues, companies in EU are starting to only allow 'western'  extensions to install on company laptop

**Comment by Yuki2718** [comment]
I seriously have no idea how urlfilter could "provide full uBO functionality".
